src/components/Popconfirm/index.tsx

```diff
@@ -1,50 +1,78 @@
-import { useRef } from "react";
+import { cloneElement, useRef, useState } from "react";
 import styled from "@emotion/styled";
 import Button from "@leafygreen-ui/button";
 import { palette } from "@leafygreen-ui/palette";
 import Popover, { Align, Justify } from "@leafygreen-ui/popover";
+import { wordBreakCss } from "components/styles";
 import { size, zIndex } from "constants/tokens";
 import { useOnClickOutside } from "hooks";
 
 const { white } = palette;
 
-interface PopconfirmProps {
+interface UncontrolledPopconfirmProps {
+  active?: never;
+  setActive?: never;
+  refEl?: never;
+  trigger: JSX.Element;
+}
+
+interface ControlledPopconfirmProps {
   active: boolean;
+  setActive: (open: boolean) => void;
+  refEl?: React.RefObject<HTMLElement>;
+  trigger?: never;
+}
+
+interface PopconfirmBase {
   align?: Align;
-  justify?: Justify;
-  "data-cy"?: string;
   children: React.ReactNode;
+  confirmDisabled?: boolean;
   confirmText?: string;
-  refEl?: React.RefObject<HTMLElement>;
+  "data-cy"?: string;
+  justify?: Justify;
   onCancel?: () => void;
-  onConfirm?: () => void;
-  setActive: (open: boolean) => void;
+  onConfirm?: (e?: React.MouseEvent) => void;
 }
 
+type PopconfirmProps = PopconfirmBase &
+  (UncontrolledPopconfirmProps | ControlledPopconfirmProps);
+
 const Popconfirm: React.VFC<PopconfirmProps> = ({
   active,
   align = "top",
   justify = "middle",
   "data-cy": dataCy,
   children,
+  confirmDisabled = false,
   confirmText = "Yes",
   refEl,
   onCancel = () => {},
   onConfirm = () => {},
   setActive,
+  trigger,
 }) => {
+  const [uncontrolledOpen, uncontrolledSetOpen] = useState<boolean>(false);
+  const setOpen =
+    (typeof active === "boolean" && setActive) || uncontrolledSetOpen;
```

Comment:
Can you explain this type check? 
couldn't you do this?
```suggestion
    (active === null" && setActive) || uncontrolledSetOpen;
```